### DISCUSSION TITLE
JavaScript send AJAX request to check if a question exists in a MySQL database and if it does render it otherwise post it

### DISCUSSION THREAD

**Discussion by ghost** [community/community#142861] [Programming Help]
### Body

```html
<!DOCTYPE html>
<html>
	<head>
    		<title>Ask a question</title>
    </head>
    <body>
        		<script>
                 //This function here will search for the question and if it cannot find it
                // server will upload it 
                function search(){
            			var question = document.getElementById("question").value;
                    
                }
            </script>
			<form action="javascript:search()">
            	<input type="text" id="question" value="" placeholder="Your question > ";
        		<input type="submit">
            </form>
    </body>
</html>
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by dalvarez2596** [comment]
@WorldsDumbestCoder  It’s important to understand which request you’re making to the backend and to ensure that your backend server is running. The backend should connect to the database, and based on the data it receives, you can check if a question exists. If it does, you can handle the response and update your UI accordingly. Please note that you need to set up a backend server to connect to the database, as direct connections from the browser are not feasible.
Please provide more info about to which service, api, etc you are connecting so we can help :)

**Reply by ghost** [reply]
GET request and PHP testing server running

**Reply by dalvarez2596** [reply]
Do you want to save the new question on the DB ?
Or only need to check if exists and render it ?
In case you want to save it, you should use a POST, but if only want to check and render it if doesn't  exists you can use fetch in your search function.
Do you a tool or something that might help doing the request ? In order to see how the params should be passed, or if theres a middleware needed.
In case is something basic you can do something like :
```

<!DOCTYPE html>
<html>
  <head>
    <title>Ask a question</title>
  </head>
  <body>
    <script>
      //This function here will search for the question and if it cannot find it
      // server will upload it
      function search() {
        var question = document.getElementById("question").value;

        fetch(`yourapiroute=${question}`)
          .then(response => response.json())
            .then(data => {
              if (data.exists) {
                console.log('Value exists in the database.');
            } else {
              console.log('Value does not exist.');
              // Here should be the place where you write the question in the  html document, i dont know maybe have another function that write the questions in a ul then iterates and add the new elements in a li
          }
          })
          .catch(error => {
            console.error('Error checking value:', error);
          });
      }
    </script>
    <form action="javascript:search()">
      <input
        type="text"
        id="question"
        value=""
        placeholder="Your question > "
        ;
        <input
        type="submit"
      />
    </form>
  </body>
</html>

```

Also you can check this https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch in order to learn more about fetch api
